lib/portlayer/exec/container.go

```diff
@@ -393,16 +393,24 @@ func infraContainers(ctx context.Context, sess *session.Session) ([]mo.VirtualMa
 
 // find the childVM for this resource pool by name
 func childVM(ctx context.Context, sess *session.Session, name string) (*vm.VirtualMachine, error) {
-	searchIndex := object.NewSearchIndex(sess.Client.Client)
-	child, err := searchIndex.FindChild(ctx, VCHConfig.ResourcePool.Reference(), name)
```

Comment:
vSphere won't rename it.  When you specify the full path to .vmx, vSphere will raise an error if the .vmx already exists.
Yes, it works for vSAN.